### PR TITLE
deposit: feature backport from analysis preservation

### PR DESCRIPTION
**Pull request by crepererum**
This contains features developed originally for the analysis preservation overlay. I think they are useful for all overlays, so I've backported them.

Rebased against: `master`

### CODE REVIEW

**Comment by crepererum** [comment]
Am I forced to find a better solution for `invenio/modules/deposit/fields/__init__.py`?

File: invenio/modules/deposit/static/css/deposit/form.css [hunk 1]
```diff
@@ -64,6 +64,42 @@ span.ui-icon.ui-icon-circle-triangle-e {
...
+.panel-toggle.collapsed .hide-on-collapsed,.show-on-collapsed{
+	display:none
+}
+
+.panel-toggle.collapsed .show-on-collapsed{
+	display:block
+}
+
+.field-without-label {
+  padding: 0 20px;
```
Comment (jirikuncar) [inline invenio/modules/deposit/static/css/deposit/form.css]: can this file be converted to less and use the constants from `variables.less`?

File: invenio/modules/deposit/templates/deposit/run_base.html [hunk 2]
```diff
@@ -47,18 +47,31 @@
 {% endblock form_submit_dialog %}
 {% block deposit_page %}
 <div class="row">
+        <div class="col-md-3 col-md-push-9">
```
Comment (jirikuncar) [inline invenio/modules/deposit/templates/deposit/run_base.html]: can you use 2 space indentation in html templates?

File: invenio/base/static/css/invenio.css [hunk 3]
```diff
@@ -1,28 +1,30 @@
...
-   You should have received a copy of the GNU General Public License
-   along with Invenio; if not, write to the Free Software Foundation, Inc.,
-   59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
-*/
 body {
    color: #000;
    background: #fff;
    margin: 0px;
    padding: 0px;
+   position: relative;
```
Comment (jirikuncar) [inline invenio/base/static/css/invenio.css]: this file is going to be removed in #2047. where do you need to change this css to be change?

File: invenio/modules/deposit/static/css/deposit/form.css (hunk 1 above)
Comment (crepererum) [inline invenio/modules/deposit/static/css/deposit/form.css]: which `variables.less`? A new one? If so: the file contains a bunch of variables that do not really belong together and that are for distinct purposes. Not really sure if it's worth to find some forced variables here. We do not have any common space/distance calculation.

**Comment by crepererum** [comment]
done

File: invenio/modules/deposit/templates/deposit/run_base.html [hunk 4]
```diff
@@ -51,6 +51,20 @@
...
             {% include "deposit/myview.html" %}
         {% endif %}
+        {% block form_navigation %}
+        <div class="sticky well scrollspy-target">
+            <ul class="nav nav-pills nav-stacked deposit-nav">
+                <li role="presentation"><a href="#deposit-top"><i class="fa fa-align-justify fa-fw"></i> {{ form._title }}</a></li>
+            {% for group, fields in form.get_groups() %}
+                {% set grouploop = loop %}
+                    <li role="presentation"><a href="#depositgroup-{{ grouploop.index }}">
+                        <i class="fa fa-dot-circle-o fa-fw"></i> {{ group.name if group.name else _('Other Information') }}{% if group.meta.indication %}<span class="text-muted pull-right" style="padding-right: 10px;">&nbsp;{{ group.meta.indication }}</span>{% endif %}
```
Comment (jirikuncar) [inline invenio/modules/deposit/templates/deposit/run_base.html]: `group.meta.indication` fails when `group is None`

File: invenio/modules/deposit/templates/deposit/run_base.html [hunk 5]
```diff
@@ -51,6 +51,20 @@
         {% if form._drafting %}
             {% include "deposit/myview.html" %}
         {% endif %}
+        {% block form_navigation %}
+        <div class="sticky well scrollspy-target">
+            <ul class="nav nav-pills nav-stacked deposit-nav">
+                <li role="presentation"><a href="#deposit-top"><i class="fa fa-align-justify fa-fw"></i> {{ form._title }}</a></li>
+            {% for group, fields in form.get_groups() %}
+                {% set grouploop = loop %}
+                    <li role="presentation"><a href="#depositgroup-{{ grouploop.index }}">
```
Comment (jirikuncar) [inline invenio/modules/deposit/templates/deposit/run_base.html]: overindented - `{% set ... %} is not block section

**Comment by crepererum** [comment]
done

File: invenio/modules/deposit/static/less/deposit/form.less [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+}
+.upload-speed {
+  margin-right: 10px;
+}
+.upload-stop {
+  display:none;
+}
+
+.hide-on-collapsed{
+	display:block
```
Comment (jirikuncar) [inline invenio/modules/deposit/static/less/deposit/form.less]: indentation?

File: invenio/modules/deposit/static/less/deposit/form.less [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+#uploader-upload {
+  display: none;
+}
+
+#uploader-stop {
+  display: none;
+}
+
+#uploader-filelist th {
+    min-width: 60px;
```
Comment (jirikuncar) [inline invenio/modules/deposit/static/less/deposit/form.less]: indentation?

**Comment by crepererum** [comment]
Sorry, that file wasn't meant to be there.

**Comment by crepererum** [comment]
done

**Pull request merged**